src/v/archival/archival_metadata_stm.cc

```diff
@@ -1030,12 +1053,23 @@ ss::future<> archival_metadata_stm::apply(const model::record_batch& b) {
                       serde::from_iobuf<update_highest_producer_id_cmd::value>(
                         r.release_value()));
                     break;
+                case read_write_fence_cmd::key:
+                    if (apply_read_write_fence(
+                          serde::from_iobuf<read_write_fence_cmd>(
+                            r.release_value()))) {
+                        // This means that there is a concurrency violation. The
+                        // fence was created before some other command was
+                        // applied. We can't apply the commands from this batch.
+                        return ss::stop_iteration::yes;
```

Comment:
nice. so is this resumable? you skip the batch, or this indicates a terrible situation?